apps/src/gamelab/AnimationTab/ListItemThumbnail.jsx

```diff
@@ -43,14 +42,34 @@ var staticStyles = {
 /**
  * Animation or Frame thumbnail.
  */
-var ListItemThumbnail = React.createClass({
+const ListItemThumbnail = React.createClass({
   propTypes: {
+    animation: React.PropTypes.shape(METADATA_SHAPE).isRequired,
     index: React.PropTypes.number,
-    isSelected: React.PropTypes.bool,
-    src: React.PropTypes.string.isRequired
+    isSelected: React.PropTypes.bool
+  },
+
+  getInitialState() {
+    return {
+      previewSize: 0
+    };
+  },
+
+  componentDidMount() {
+    this.recalculatePreviewSize();
+  },
+
+  forceResize() {
+    this.recalculatePreviewSize();
+  },
```

Comment:
I'll annotate this `@public` until we make a decision.  It'd be nice to get away from the underscore notation.